torch/_inductor/constant_folding.py

```diff
@@ -109,6 +109,13 @@ def set_env(arg):
         ):
             return self.unknown_value
 
+        # TODO - constant folding triton kernel returns the inputs -- fix this
```

Comment:
`# TODO - constant folding triton kernel returns the inputs -- fix this`

What is the TODO here exactly ?

Reply:
I guess it is to evaluate whether we can in fact constant fold or not. Since it is returning the inputs, i suspect the fake tensor dispatcher is being used here but intuitively, it should just be `ProxyTorchDispatchMode`

Reply:
Constant Folding is running the ops directly, it shouldnt be using proxy or fake tensor. 

Could we add test ? Also, do we generate a schema for these ops

Reply:
The test I added to this PR was running into both constant folding and externel kernel out mutation tracking bugs.
The constant folding was happing in joint graph passes. Did you want a different test?

Reply:
Okay nevermind, it doesn't have a schema then.